.changeset/pink-baths-marry.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/schema-compat': patch
+---
+
+Fixed declaration imports to resolve sibling files before same-named directories and preserve copied CommonJS and ES module type dependencies.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)